lib/std/process.zig

```diff
@@ -299,196 +301,279 @@ pub const ArgIteratorWasi = struct {
     }
 };
 
-pub const ArgIteratorWindows = struct {
-    index: usize,
-    cmd_line: [*]const u16,
+/// Optional parameters for `ArgIteratorGeneral`
+pub const ArgIteratorGeneralOptions = struct {
+    comments_supported: bool = false,
+};
 
-    pub const NextError = error{ OutOfMemory, InvalidCmdLine };
+/// A general Iterator to parse a string into a set of arguments
+pub fn ArgIteratorGeneral(comptime options: ArgIteratorGeneralOptions) type {
+    return struct {
+        allocator: Allocator,
+        index: usize = 0,
+        cmd_line: [:0]const u8,
+
+        start: usize = 0,
+        end: usize = 0,
+        buffer: []u8,
+
+        pub const Self = @This();
+
+        pub const InitError = error{OutOfMemory};
+        pub const InitUtf16leError = error{ OutOfMemory, InvalidCmdLine };
+
+        /// cmd_line_utf8 is duplicated into an internal buffer
+        pub fn init(allocator: Allocator, cmd_line_utf8: []const u8) InitError!Self {
+            var cmd_line = try allocator.dupeZ(u8, cmd_line_utf8);
+            errdefer allocator.free(cmd_line);
+            var buffer = try allocator.alloc(u8, cmd_line.len + 1);
+            errdefer allocator.free(buffer);
+
+            return Self{
+                .allocator = allocator,
+                .cmd_line = cmd_line,
+                .buffer = buffer,
+            };
+        }
 
-    pub fn init() ArgIteratorWindows {
-        return initWithCmdLine(os.windows.kernel32.GetCommandLineW());
-    }
+        /// Initialize the arguments from a Response File. "*.rsp"
+        pub fn initFromResponseFile(allocator: Allocator, resp_file_path: []const u8) !Self {
```

Comment:
Let's move this logic back over to src/main.zig instead of having it here in std/process.zig.